paragon/theme.js

```diff
@@ -0,0 +1,280 @@
+const applyTheme = (oldTheme) => {
```

Comment:
Hmm, I expect that this manual deep merge could actually be done in `ParagonProvider` where it's setting up its `ThemeProvider` components, rather than here.  That way the theme itself would just be a sparse object, minus the merging logic.